lib/Main.php

```diff
@@ -2464,7 +2466,7 @@ public function uglyLink( $type = 'home', $secondary = false ) {
 					break;
 			}
 
-			return apply_filters( 'tribe_events_ugly_link', $eventUrl, $type, $secondary );
+			return apply_filters( 'tribe_events_ugly_link', esc_url( $eventUrl ), $type, $secondary );
```

Comment:
This function is called from https://github.com/moderntribe/the-events-calendar/blob/develop/lib/Main.php#L2345, which already escapes this value in a less destructive way.  Which means this probably is a bad line to escape the URL on.